vignettes/profiling.Rmd

```diff
@@ -0,0 +1,68 @@
+---
+title: "Profiling Performance"
+author: "Thomas Lin Pedersen"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Profiling Performance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+The performance of `gtable` is critical as it has the potential to slow down
+plot creation a lot. In order to continuously monitor the performance of gtable
+the following piece of code is used to generate a profiling and inspect it:
+
+```{r}
+library(ggplot2)
+library(profvis)
+
+p <- ggplot() + 
+  scale_x_continuous(limits = c(0, 1)) + 
+  scale_y_continuous(limits = c(0, 1))
```

Comment:
Wouldn't it be better to include some facetting too? There's that not much gtable assembly for a simple plot.